client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/SutController.java

```diff
@@ -685,6 +690,29 @@ public final DbInfoDto getSqlDatabaseSchema() {
             return schemaDto;
         }
 
+        DbInfoDto schema = extractSqlDbSchemaAndConstraints();
+        if (schema == null) return null;
+
+        UnitsInfoDto unitsInfoDto = getUnitsInfoDto();
+        List<ExtraConstraintsDto> extra = unitsInfoDto.extraDatabaseConstraintsDtos;
+        if( extra != null && !extra.isEmpty()) {
+            schemaDto.extraConstraintDtos = new ArrayList<>();
+            schemaDto.extraConstraintDtos.addAll(extra);
+        }
+
+        return schemaDto;
+    }
+
+    /**
+     * @return extracted Sql Database Schema. Note that null is returned only if the schema has been extracted or the extraction fails with the given db connection.
+     */
+    public final DbInfoDto extractSqlDbSchemaAndConstraints(){
```

Comment:
this is bit confusing... it is a public method. if it is called with an existing `schemaDto` it will return `null`?

Reply:
this function has been changed to return `Boolean` representing whether the extraction has been performed successfully.